cx_Freeze/dist.py

```diff
@@ -236,7 +237,18 @@ def run(self):
             warnings.warn(
                 "namespace-packages is obsolete and will be removed in the next version"
             )
-        freezer = Freezer(
+
+        freeze_class: Type[Freezer]
+        if sys.platform == "win32":
+            freeze_class = WinFreezer
+        elif sys.platform == "darwin":
+            freeze_class = DarwinFreezer
+        elif sys.platform == "linux":
+            freeze_class = LinuxFreezer
+        else:
+            raise Exception(f'Unknown platform: {sys.platform}')
+
+        freezer: Freezer = freeze_class(
```

Comment:
I think that this should be moved back to freeze.py, so the API maintains compatiblilty. While this is being tested, you can leave it at that.

Reply:
Okay, that's easy enough to fix. We can make it so that the existence of the subclasses is transparent for someone using the Freezer constructor.  (just made a commit to do that)